packages/mui-base/src/useSwitch/useSwitch.ts

```diff
@@ -90,17 +78,15 @@ export function useSwitch(props: UseSwitchParameters): UseSwitchReturnValue {
   const createHandleBlur =
     (otherProps: React.InputHTMLAttributes<HTMLInputElement>) =>
     (event: React.FocusEvent<HTMLInputElement>) => {
-      handleBlurVisible(event);
-
-      if (isFocusVisibleRef.current === false) {
+      if (!isFocusVisible(event.target)) {
         setFocusVisible(false);
       }
 
       onBlur?.(event);
       otherProps.onBlur?.(event);
     };
 
-  const handleInputRef = useForkRef(focusVisibleRef, inputRef);
+  const handleInputRef = useForkRef(inputRef);
```

Comment:
Is this `useForkRef` needed?

Reply:
The `ref` argument in `UseSwitchReturnValue` is typed to match `useForkRef`'s return, which is different from `inputRef`'s type. I would rather not change this mui-base code 😅

Reply:
Reference: https://github.com/mui/material-ui/blob/next/packages/mui-base/src/useSwitch/useSwitch.types.ts#L77

Reply:
Makes sense